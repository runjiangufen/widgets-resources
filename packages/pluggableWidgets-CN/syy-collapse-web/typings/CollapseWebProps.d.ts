/**
 * This file was generated from CollapseWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";

export type CollapsibleEnum = "header" | "disabled";

export type ExpandIconPositionEnum = "start" | "end";

export interface CollapseWebContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    children?: ReactNode;
    accordion: boolean;
    bordered: boolean;
    collapsible: CollapsibleEnum;
    expandIconPosition: ExpandIconPositionEnum;
    ghost: boolean;
    onChangeAction?: ActionValue;
}

export interface CollapseWebPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    children: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    accordion: boolean;
    bordered: boolean;
    collapsible: CollapsibleEnum;
    expandIconPosition: ExpandIconPositionEnum;
    ghost: boolean;
    onChangeAction: {} | null;
}
