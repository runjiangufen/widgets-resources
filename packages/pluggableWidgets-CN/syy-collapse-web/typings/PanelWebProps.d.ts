/**
 * This file was generated from PanelWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";

export type CollapsibleEnum = "header" | "disabled" | "undefined";

export interface PanelWebContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    children?: ReactNode;
    key: string;
    collapsible: CollapsibleEnum;
    extra?: ReactNode;
    forceRender: boolean;
    header: ReactNode;
    showArrow: boolean;
}

export interface PanelWebPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    children: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    key: string;
    collapsible: CollapsibleEnum;
    extra: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    forceRender: boolean;
    header: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    showArrow: boolean;
}
