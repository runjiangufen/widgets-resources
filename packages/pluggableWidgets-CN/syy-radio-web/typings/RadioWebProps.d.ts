/**
 * This file was generated from RadioWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue, ListValue, ListAttributeValue } from "mendix";
import { Big } from "big.js";

export type DatasTypeEnum = "dynamic" | "static";

export interface StaticDataType {
    manualValue: string;
    manuaTitle: string;
    manualDisabled: boolean;
}

export type SizeEnum = "large" | "middle" | "small";

export type ButtonStyleEnum = "outline" | "solid";

export type OptionTypeEnum = "default" | "button";

export type DirectionEnum = "horizontal" | "vertical";

export interface StaticDataPreviewType {
    manualValue: string;
    manuaTitle: string;
    manualDisabled: boolean;
}

export interface RadioWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    datasType: DatasTypeEnum;
    staticData: StaticDataType[];
    data?: ListValue;
    attribute?: ListAttributeValue<string | Date | Big>;
    title?: ListAttributeValue<string>;
    attributedata?: EditableValue<string>;
    selectedValue?: EditableValue<string | Date | Big>;
    isText: boolean;
    size: SizeEnum;
    buttonStyle: ButtonStyleEnum;
    disabled: DynamicValue<boolean>;
    radioName: string;
    optionType: OptionTypeEnum;
    direction: DirectionEnum;
    onChangeAction?: ActionValue;
    customWidgetOnChangeValidate: boolean;
}

export interface RadioWebPreviewProps {
    readOnly: boolean;
    datasType: DatasTypeEnum;
    staticData: StaticDataPreviewType[];
    data: {} | { type: string } | null;
    attribute: string;
    title: string;
    attributedata: string;
    selectedValue: string;
    isText: boolean;
    size: SizeEnum;
    buttonStyle: ButtonStyleEnum;
    disabled: string;
    radioName: string;
    optionType: OptionTypeEnum;
    direction: DirectionEnum;
    onChangeAction: {} | null;
    customWidgetOnChangeValidate: boolean;
}
