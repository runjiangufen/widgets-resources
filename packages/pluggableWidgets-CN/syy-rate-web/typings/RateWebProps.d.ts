/**
 * This file was generated from RateWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue } from "mendix";
import { Big } from "big.js";

export interface RateWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    value: EditableValue<Big>;
    allowClear: boolean;
    allowHalf: boolean;
    count: number;
    disabled: DynamicValue<boolean>;
    onChange?: ActionValue;
    customWidgetOnChangeValidate: boolean;
}

export interface RateWebPreviewProps {
    readOnly: boolean;
    value: string;
    allowClear: boolean;
    allowHalf: boolean;
    count: number | null;
    disabled: string;
    onChange: {} | null;
    customWidgetOnChangeValidate: boolean;
}
