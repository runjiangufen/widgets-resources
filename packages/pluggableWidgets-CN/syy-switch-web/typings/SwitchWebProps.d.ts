/**
 * This file was generated from SwitchWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue } from "mendix";

export type SizeEnum = "default" | "small";

export type ConfirmTypeEnum = "pop" | "modal";

export type ConfirmokTypeEnum = "primary" | "ghost" | "dashed" | "link" | "text" | "default";

export interface SwitchWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    checked: EditableValue<boolean>;
    onName: string;
    offName: string;
    loadingFlag?: EditableValue<boolean>;
    disabled?: EditableValue<boolean>;
    size: SizeEnum;
    autoFocus: boolean;
    onChange?: ActionValue;
    openconfirm: boolean;
    confirmType: ConfirmTypeEnum;
    ontitle?: DynamicValue<string>;
    offtitle?: DynamicValue<string>;
    confirmokText: string;
    confirmcancelText: string;
    confirmokType: ConfirmokTypeEnum;
    onConfirm?: ActionValue;
    onCancel?: ActionValue;
}

export interface SwitchWebPreviewProps {
    readOnly: boolean;
    checked: string;
    onName: string;
    offName: string;
    loadingFlag: string;
    disabled: string;
    size: SizeEnum;
    autoFocus: boolean;
    onChange: {} | null;
    openconfirm: boolean;
    confirmType: ConfirmTypeEnum;
    ontitle: string;
    offtitle: string;
    confirmokText: string;
    confirmcancelText: string;
    confirmokType: ConfirmokTypeEnum;
    onConfirm: {} | null;
    onCancel: {} | null;
}
