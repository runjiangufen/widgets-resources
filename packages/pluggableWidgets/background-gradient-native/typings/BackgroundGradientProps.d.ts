/**
 * This file was generated from BackgroundGradient.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue } from "mendix";
import { Big } from "big.js";

export interface ColorListType {
    color: string;
    offset: Big;
}

export interface ColorListPreviewType {
    color: string;
    offset: number | null;
}

export interface BackgroundGradientProps<Style> {
    name: string;
    style: Style[];
    colorList: ColorListType[];
    content: ReactNode;
    onClick?: ActionValue;
}

export interface BackgroundGradientPreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    colorList: ColorListPreviewType[];
    content: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    onClick: {} | null;
}
