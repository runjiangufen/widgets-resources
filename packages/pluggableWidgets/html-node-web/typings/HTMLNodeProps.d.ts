/**
 * This file was generated from HTMLNode.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ComponentType, CSSProperties, ReactNode } from "react";
import { ActionValue, DynamicValue, ListValue, ListActionValue, ListExpressionValue, ListWidgetValue } from "mendix";

export type TagNameEnum =
    | "div"
    | "span"
    | "p"
    | "ul"
    | "ol"
    | "li"
    | "a"
    | "img"
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "__customTag__";

export type AttributeValueTypeEnum = "expression" | "template";

export interface AttributesType {
    attributeName: string;
    attributeValueType: AttributeValueTypeEnum;
    attributeValueTemplate?: DynamicValue<string>;
    attributeValueExpression?: DynamicValue<string>;
    attributeValueTemplateRepeat?: ListExpressionValue<string>;
    attributeValueExpressionRepeat?: ListExpressionValue<string>;
}

export type TagContentModeEnum = "container" | "innerHTML";

export type EventNameEnum =
    | "onClick"
    | "onChange"
    | "onFocus"
    | "onLeave"
    | "onSubmit"
    | "onKeyDown"
    | "onKeyUp"
    | "onCopy"
    | "onCopyCapture"
    | "onCut"
    | "onCutCapture"
    | "onPaste"
    | "onPasteCapture"
    | "onCompositionEnd"
    | "onCompositionEndCapture"
    | "onCompositionStart"
    | "onCompositionStartCapture"
    | "onCompositionUpdate"
    | "onCompositionUpdateCapture"
    | "onFocusCapture"
    | "onBlur"
    | "onBlurCapture"
    | "onChangeCapture"
    | "onBeforeInput"
    | "onBeforeInputCapture"
    | "onInput"
    | "onInputCapture"
    | "onReset"
    | "onResetCapture"
    | "onSubmitCapture"
    | "onInvalid"
    | "onInvalidCapture"
    | "onLoad"
    | "onLoadCapture"
    | "onError"
    | "onErrorCapture"
    | "onKeyDownCapture"
    | "onKeyPress"
    | "onKeyPressCapture"
    | "onKeyUpCapture"
    | "onAbort"
    | "onAbortCapture"
    | "onCanPlay"
    | "onCanPlayCapture"
    | "onCanPlayThrough"
    | "onCanPlayThroughCapture"
    | "onDurationChange"
    | "onDurationChangeCapture"
    | "onEmptied"
    | "onEmptiedCapture"
    | "onEncrypted"
    | "onEncryptedCapture"
    | "onEnded"
    | "onEndedCapture"
    | "onLoadedData"
    | "onLoadedDataCapture"
    | "onLoadedMetadata"
    | "onLoadedMetadataCapture"
    | "onLoadStart"
    | "onLoadStartCapture"
    | "onPause"
    | "onPauseCapture"
    | "onPlay"
    | "onPlayCapture"
    | "onPlaying"
    | "onPlayingCapture"
    | "onProgress"
    | "onProgressCapture"
    | "onRateChange"
    | "onRateChangeCapture"
    | "onSeeked"
    | "onSeekedCapture"
    | "onSeeking"
    | "onSeekingCapture"
    | "onStalled"
    | "onStalledCapture"
    | "onSuspend"
    | "onSuspendCapture"
    | "onTimeUpdate"
    | "onTimeUpdateCapture"
    | "onVolumeChange"
    | "onVolumeChangeCapture"
    | "onWaiting"
    | "onWaitingCapture"
    | "onAuxClick"
    | "onAuxClickCapture"
    | "onClickCapture"
    | "onContextMenu"
    | "onContextMenuCapture"
    | "onDoubleClick"
    | "onDoubleClickCapture"
    | "onDrag"
    | "onDragCapture"
    | "onDragEnd"
    | "onDragEndCapture"
    | "onDragEnter"
    | "onDragEnterCapture"
    | "onDragExit"
    | "onDragExitCapture"
    | "onDragLeave"
    | "onDragLeaveCapture"
    | "onDragOver"
    | "onDragOverCapture"
    | "onDragStart"
    | "onDragStartCapture"
    | "onDrop"
    | "onDropCapture"
    | "onMouseDown"
    | "onMouseDownCapture"
    | "onMouseEnter"
    | "onMouseLeave"
    | "onMouseMove"
    | "onMouseMoveCapture"
    | "onMouseOut"
    | "onMouseOutCapture"
    | "onMouseOver"
    | "onMouseOverCapture"
    | "onMouseUp"
    | "onMouseUpCapture"
    | "onSelect"
    | "onSelectCapture"
    | "onTouchCancel"
    | "onTouchCancelCapture"
    | "onTouchEnd"
    | "onTouchEndCapture"
    | "onTouchMove"
    | "onTouchMoveCapture"
    | "onTouchStart"
    | "onTouchStartCapture"
    | "onPointerDown"
    | "onPointerDownCapture"
    | "onPointerMove"
    | "onPointerMoveCapture"
    | "onPointerUp"
    | "onPointerUpCapture"
    | "onPointerCancel"
    | "onPointerCancelCapture"
    | "onPointerEnter"
    | "onPointerEnterCapture"
    | "onPointerLeave"
    | "onPointerLeaveCapture"
    | "onPointerOver"
    | "onPointerOverCapture"
    | "onPointerOut"
    | "onPointerOutCapture"
    | "onGotPointerCapture"
    | "onGotPointerCaptureCapture"
    | "onLostPointerCapture"
    | "onLostPointerCaptureCapture"
    | "onScroll"
    | "onScrollCapture"
    | "onWheel"
    | "onWheelCapture"
    | "onAnimationStart"
    | "onAnimationStartCapture"
    | "onAnimationEnd"
    | "onAnimationEndCapture"
    | "onAnimationIteration"
    | "onAnimationIterationCapture"
    | "onTransitionEnd"
    | "onTransitionEndCapture";

export interface EventsType {
    eventName: EventNameEnum;
    eventAction?: ActionValue;
    eventActionRepeat?: ListActionValue;
    eventStopPropagation: boolean;
    eventPreventDefault: boolean;
}

export interface AttributesPreviewType {
    attributeName: string;
    attributeValueType: AttributeValueTypeEnum;
    attributeValueTemplate: string;
    attributeValueExpression: string;
    attributeValueTemplateRepeat: string;
    attributeValueExpressionRepeat: string;
}

export interface EventsPreviewType {
    eventName: EventNameEnum;
    eventAction: {} | null;
    eventActionRepeat: {} | null;
    eventStopPropagation: boolean;
    eventPreventDefault: boolean;
}

export interface HTMLNodeContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    tagName: TagNameEnum;
    tagNameCustom: string;
    attributes: AttributesType[];
    tagUseRepeat: boolean;
    tagContentRepeatDataSource?: ListValue;
    tagContentMode: TagContentModeEnum;
    tagContentHTML?: DynamicValue<string>;
    tagContentContainer?: ReactNode;
    tagContentRepeatHTML?: ListExpressionValue<string>;
    tagContentRepeatContainer?: ListWidgetValue;
    events: EventsType[];
}

export interface HTMLNodePreviewProps {
    className: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    tagName: TagNameEnum;
    tagNameCustom: string;
    attributes: AttributesPreviewType[];
    tagUseRepeat: boolean;
    tagContentRepeatDataSource: {} | { type: string } | null;
    tagContentMode: TagContentModeEnum;
    tagContentHTML: string;
    tagContentContainer: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    tagContentRepeatHTML: string;
    tagContentRepeatContainer: { widgetCount: number; renderer: ComponentType<{ caption?: string }> };
    events: EventsPreviewType[];
}
