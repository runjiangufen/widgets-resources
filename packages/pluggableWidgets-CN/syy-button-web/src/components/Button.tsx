import { ReactElement, createElement, ReactNode } from "react";
import { PopConpemnet, PopConpemnetProps } from "./PopConpemnet";
import { ButtonConpemnet, ButtonConpemnetProps } from "./ButtonConpemnet";

export interface ButtonWebProps extends ButtonConpemnetProps, PopConpemnetProps {
    children: ReactNode;
    openconfirm?: boolean;
    confirmType?: "pop" | "modal";
}
// ButtonConpemnet
export function Button(props: ButtonWebProps): ReactElement {
    const {
        block,
        href,
        ghost,
        loading,
        danger,
        disabled,
        size,
        icon,
        shape,
        type,
        children,
        openconfirm, // Open the confirmation pop-up window before executing microflow
        confirmType,
        confirmtitle,
        confirmokText,
        onClick,
        onConfirm,
        onCancel,
        confirmcancelText,
        confirmokType
    } = props;

    const ButtonWebView = (
        <ButtonConpemnet
            disabled={disabled}
            danger={danger}
            loading={loading}
            ghost={ghost}
            block={block}
            href={href}
            size={size}
            shape={shape}
            type={type}
            onClick={onClick}
            icon={icon}
        >
            {children}
        </ButtonConpemnet>
    );
    return (
        <div>
            {openconfirm && confirmType === "pop" ? (
                <PopConpemnet
                    confirmtitle={confirmtitle}
                    confirmokText={confirmokText}
                    confirmcancelText={confirmcancelText}
                    confirmokType={confirmokType}
                    onConfirm={onConfirm}
                    onCancel={onCancel}
                >
                    {ButtonWebView}
                </PopConpemnet>
            ) : (
                ButtonWebView
            )}
        </div>
    );
}
