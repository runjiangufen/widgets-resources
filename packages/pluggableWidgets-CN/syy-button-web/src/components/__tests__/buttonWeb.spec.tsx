import { createElement } from "react";
import { render, mount } from "enzyme";
import { Button as ButtonConpemnet, ButtonWebProps } from "../Button";
import { ConfirmTypeEnum, ConfirmokTypeEnum } from "../../../typings/ButtonWebProps";

interface popConfirmButtonProps {
    openconfirm: boolean;
    confirmType?: ConfirmTypeEnum;
    confirmtitle?: string;
    confirmokText?: string;
    confirmcancelText?: string;
    confirmokType?: ConfirmokTypeEnum;
    onConfirm?: () => void;
    onCancel?: () => void;
}
describe("ButtonWeb", () => {
    let defaultButtonProps: ButtonWebProps;
    let popConfirmButtonProps: popConfirmButtonProps;

    beforeEach(() => {
        defaultButtonProps = {
            onClick: undefined,
            type: "default",
            openconfirm: false,
            children: "Button",
            icon: { type: "glyph", iconClass: "icon-class" }
        };

        popConfirmButtonProps = {
            openconfirm: true,
            confirmType: "pop",
            confirmtitle: "Are you sure to delete this task?",
            confirmokText: "Yes",
            confirmcancelText: "No",
            confirmokType: "primary",
            onConfirm: jest.fn(),
            onCancel: jest.fn()
        };
    });

    it("renders default buttons", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} />);
        expect(buttonWrapper).toMatchSnapshot();
    });

    it("Set the type of button", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} type="primary" />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("Set the size of button", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} size="middle" />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("Disabled state of button", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} disabled />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("Set the loading status of button", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} loading />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("Make background transparent and invert text and border colors", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} ghost />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("Set the danger status of button", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} danger />);
        expect(buttonWrapper).toMatchSnapshot();
    });

    it("Option to fit button width to its parent width", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} block />);
        expect(buttonWrapper).toMatchSnapshot();
    });

    it("Redirect url of link button", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} type="link" href="#" />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("Can be set button shape", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} shape="circle" />);
        expect(buttonWrapper).toMatchSnapshot();
    });

    it("compact dialog used for asking for user confirmation", () => {
        const buttonWrapper = render(<ButtonConpemnet {...defaultButtonProps} {...popConfirmButtonProps} />);
        expect(buttonWrapper).toMatchSnapshot();
    });
    it("modal dialog used for asking for user confirmation", () => {
        const buttonWrapper = render(
            <ButtonConpemnet {...defaultButtonProps} {...popConfirmButtonProps} confirmType="modal" />
        );
        expect(buttonWrapper).toMatchSnapshot();
    });

    it("triggers onClick function with a click event", () => {
        const onClick = jest.fn();
        const buttonWrapper = mount(<ButtonConpemnet {...defaultButtonProps} onClick={onClick} />);
        const button = buttonWrapper.find("button");
        button.simulate("click");
        expect(onClick).toHaveBeenCalledTimes(1);
    });
});
