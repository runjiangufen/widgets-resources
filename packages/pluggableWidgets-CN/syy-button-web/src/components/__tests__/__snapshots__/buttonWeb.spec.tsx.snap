// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWeb Can be set button shape 1`] = `
<div>
  <button
    class="ant-btn ant-btn-circle ant-btn-default"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Disabled state of button 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default"
    disabled=""
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Make background transparent and invert text and border colors 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default ant-btn-background-ghost"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Option to fit button width to its parent width 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default ant-btn-block"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Redirect url of link button 1`] = `
<div>
  <a
    class="ant-btn ant-btn-link"
    href="#"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </a>
</div>
`;

exports[`ButtonWeb Set the danger status of button 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default ant-btn-dangerous"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Set the loading status of button 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default ant-btn-loading"
    type="button"
  >
    <span
      class="ant-btn-loading-icon"
    >
      <span
        aria-label="loading"
        class="anticon anticon-loading anticon-spin"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </span>
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Set the size of button 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb Set the type of button 1`] = `
<div>
  <button
    class="ant-btn ant-btn-primary"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb compact dialog used for asking for user confirmation 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb modal dialog used for asking for user confirmation 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`ButtonWeb renders default buttons 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span
      aria-hidden="true"
      class="glyphicon rating-icon rating-icon-empty icon-class"
    />
    <span>
      Button
    </span>
  </button>
</div>
`;
